### DISCUSSION TITLE
Groups with NumPad Keys

### DISCUSSION THREAD

**Discussion by DasLixou** [qtile/qtile#3735] [Q&A] [answered]
Hi! I got the default configuration from installing qtile and wondered what I must to so that numpadkeys also work to switch groups. Do I only have to copy paste the extending keys and add smth like "NumKey" + i ? Thanks :)

**Comment by m-col** [comment, marked as answer]
I'm not 100% certain but I suspect will have dedicated names, you can check these by opening `xev` and pressing those keys, xev will print the key names

**Reply by elParaguayo** [reply]
I think they're `KP_0`, `KP_1` etc. but that may depend whether numlock is on or not (my laptop doesn't have a number pad...). `xev` should confirm.

**Reply by DasLixou** [reply]
Xev says KP_Begin KP_End KP_Up and so. Is there a way to get these from numbers?

**Reply by elParaguayo** [reply]
I assume you're trying to use the loop in the config to add keybindings, in which case, you could do something like this:
``` python
groups = [Group(i) for i in "123456789"]

numpad_keys = ['KP_Begin', 'KP_End'...] 

for i, numpad_key in zip (groups, numpad_keys):
    keys.extend(
        [
            Key(
                [mod],
                numpad_key
                lazy.group[i.name].toscreen(),
                desc="Switch to group {}".format(i.name),
            ),

            Key(
                [mod, "shift"],
                numpad_key,
                lazy.window.togroup(i.name, switch_group=True),
                desc="Switch to & move focused window to group {}".format(i.name),
            ),
        ]
    )
```
